src/lib/util.ts

```diff
@@ -80,3 +95,53 @@ export async function canvasEncode(data: ImageData, type: string, quality?: numb
   if (!blob) throw Error('Encoding failed');
   return blob;
 }
+
+export function canDecodeImage(data: string): Promise<boolean> {
+  return new Promise((resolve) => {
+    const img = document.createElement('img');
+    img.src = data;
+    img.onload = _ => resolve(true);
+    img.onerror = _ => resolve(false);
+  });
+}
+
+export function blobToArrayBuffer(blob: Blob): Promise<ArrayBuffer> {
+  return new Promise((resolve) => {
+    const fileReader = new FileReader();
+    fileReader.addEventListener('load', () => {
+      resolve(fileReader.result);
+    });
+    fileReader.readAsArrayBuffer(blob);
+  });
+}
+
+const magicNumberToMimeType = new Map<RegExp, string>([
+  [/^%PDF-/, 'application/pdf'],
+  [/^GIF87a/, 'image/gif'],
+  [/^GIF89a/, 'image/gif'],
+  [/^\x89PNG\x0D\x0A\x1A\x0A/, 'image/png'],
+  [/^\xFF\xD8\xFF/, 'image/jpeg'],
+  [/^BM/, 'image/bmp'],
+  [/^I I/, 'image/tiff'],
+  [/^II*/, 'image/tiff'],
+  [/^MM\x00*/, 'image/tiff'],
+  [/^RIFF....WEBPVP8 /, 'image/webp'],
+]);
+
+export async function sniffMimeType(blob: Blob): Promise<string | ''> {
```

Comment:
`''` is also a string right?

Reply:
...

Reply:
My point is that `string | ''` is redundant compared to just `string`.

Reply:
I know. I just don’t like being exposed for the idiot that I am

Reply:
😀 it's one of those mistakes that looks stupid, but given the evolution of the code you can see how it happened. Someone once pulled me up on some code like this:

```ts
const result: boolean = whatever();
if (result) {
  return true;
} else {
  return false;
}
```

"Why not just return the result of `whatever()`" haha yeah oops, I *was* doing stuff in that `if` until a refactor.